editor/effects.js

```diff
@@ -210,4 +221,28 @@ export default {
 			]
 		) );
 	},
+	AUTOSAVE( action, store ) {
+		const { dispatch, getState } = store;
+		const state = getState();
+		if ( ! isEditedPostSaveable( state ) || ! isEditedPostDirty( state ) ) {
+			return;
+		}
+
+		if ( isEditedPostPublished( state ) ) {
+			// TODO: Publish autosave
```

Comment:
>What's the desired behaviour here?

Published posts have a distinct auto-save behavior which is not currently possible to implement due to lacking API support: https://github.com/WordPress/gutenberg/issues/1773#issuecomment-316434091